### PR TITLE
Navigation: Add descriptions to NavModelItems

### LINKED ISSUE #55616: Navigation: Add description to all pages

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- initial start at adding some descriptions. settled on using `Description`, we can remove `Subtitle` post topnav release

| before | after |
| --- | --- |
| image | image |
| image | image |
| image | image |

**Which issue(s) this PR fixes**:

Fixes #55616 

**Special notes for your reviewer**:

tried to take most descriptions from the docs here: https://grafana.com/docs/grafana/latest/

### CODE REVIEW

**Review by lpskdl — APPROVED** [review]
lgtm 🚀

File: public/app/core/components/PageNew/PageHeader.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export interface Props {
 
 export function PageHeader({ navItem, subTitle }: Props) {
   const styles = useStyles2(getStyles);
-  const sub = subTitle ?? navItem.subTitle;
+  const sub = subTitle ?? navItem.description;
```
Comment (lpskdl) [inline public/app/core/components/PageNew/PageHeader.tsx:14]: I would like to ask the case where we are going to use `subTitle` ? is there a case that `navItem.description` will not be available?

Comment (ashharrison90) [reply public/app/core/components/PageNew/PageHeader.tsx:14]: naw, i want to consolidate all our mixed uses of `subTitle`/`description` to just `description` and we can remove `subTitle`.

**Review by eledobleefe — APPROVED** [review]
LGTM 👍

**Comment by ashharrison90** [comment]
gonna merge this - happy to make any changes, just want to get **something** in for now (it's all behind the feature toggle anyway) 😅

**Pull request merged**